src/app/upload/shared/upload.service.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * upload.service.ts
+ * Service providing upload.
+ *
+   Copyright 2018 Sebastian Pustelnik
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import { Injectable } from '@angular/core';
+import { Http, Headers } from '@angular/http';
+import { Observable } from 'rxjs/Rx';
+import 'rxjs/Rx';
+
+import { Service } from '../../app.service';
+import { apiUrl } from '../../../environments/apiUrl';
+
+@Injectable()
+export class UploadService extends Service {
+
+  constructor(private http: Http) {
+    super();
+  }
+
+  uploadData(datasetLink: string, datasetName): Observable<any> {
+    return this.http
+      .post(this.getBaseUrl() + apiUrl.uploadUrl, {headers: this.getHeaders(), url: datasetLink, datasetName: datasetName})
```

Comment:
`apiUrl` is not necessarily placed on `BaseUrl`. Right now we have separate ports for this, but this usually means separate hosts in fact.